# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../embutils/which-wd.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

Fixed which to inject a . when nothing is found between colons, so current
working directory binaries are located.

  - Rene Rebe <rene@exactcode.de>

--- embutils-0.17/which.c	2002-05-27 17:23:26.000000000 +0200
+++ embutils-0.17-free/which.c	2005-02-14 17:08:41.453217816 +0100
@@ -32,6 +32,7 @@
 	struct stat ss;
 	char *bar=(char*)alloca(strlen(tmp)+strlen(argv[i])+5);
 	int len=str_copy(bar,tmp);
+	if (len==0) { *bar='.'; len++; }
 	bar[len]='/'; ++len;
 	len+=str_copy(bar+len,argv[i]);
 	bar[len]=0;
